diff --git a/src/config_os.h b/src/config_os.h
index ed8e52c..5bbf7e0 100644
--- a/config_os.h
+++ b/config_os.h
@@ -29,9 +29,9 @@
 // https://www.cryptopp.com/wiki/Release_Process#Self_Tests
 // The problems with Clang pretending to be other compilers is
 // discussed at http://github.com/weidai11/cryptopp/issues/147.
-#if (defined(_MSC_VER) && defined(__clang__))
-# error: "Unsupported configuration"
-#endif
+//#if (defined(_MSC_VER) && defined(__clang__))
+//# error: "Unsupported configuration"
+//#endif
 
 // Windows platform
 #if defined(_WIN32) || defined(_WIN64) || defined(__CYGWIN__)
diff --git a/src/config_ver.h b/src/config_ver.h
index 8765844..6d054d5 100644
--- a/config_ver.h
+++ b/config_ver.h
@@ -82,7 +82,7 @@
 # define CRYPTOPP_INTEL_VERSION (__INTEL_COMPILER)
 #endif
 
-#if defined(_MSC_VER) && !defined(__clang__)
+#if defined(_MSC_VER)
 # undef CRYPTOPP_LLVM_CLANG_VERSION
 # define CRYPTOPP_MSC_VERSION (_MSC_VER)
 #endif
diff --git a/src/cryptopp/CMakeLists.txt b/src/cryptopp/CMakeLists.txt
index 7a78807..c18634d 100644
--- a/cryptopp/CMakeLists.txt
+++ b/cryptopp/CMakeLists.txt
@@ -428,7 +428,7 @@ if(
         OR CRYPTOPP_MINGW64
         OR CRYPTOPP_CYGWIN
     )
-    AND NOT MSVC
+    AND (NOT MSVC OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
 )
     if(CMAKE_CXX_COMPILER_ID STREQUAL "SunPro")
         set(sse2_flag "-xarch=sse2")
diff --git a/src/rijndael.cpp b/src/rijndael.cpp
index 3d6665e..40ca29a 100644
--- a/rijndael.cpp
+++ b/rijndael.cpp
@@ -109,7 +109,7 @@ NAMESPACE_BEGIN(CryptoPP)
 #define CONST_M128I_CAST(x) ((const __m128i *)(const void *)(x))
 
 #if defined(CRYPTOPP_ALLOW_RIJNDAEL_UNALIGNED_DATA_ACCESS)
-# if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE)) && !defined(CRYPTOPP_DISABLE_RIJNDAEL_ASM)
+# if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE))
 namespace rdtable {CRYPTOPP_ALIGN_DATA(16) word64 Te[256+2];}
 using namespace rdtable;
 # else
@@ -285,7 +285,7 @@ void Rijndael::Base::FillEncTable()
 		}
 #endif
 	}
-#if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE)) && !defined(CRYPTOPP_DISABLE_RIJNDAEL_ASM)
+#if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE))
 	Te[256] = Te[257] = 0;
 #endif
 	s_TeFilled = true;
@@ -374,7 +374,7 @@ std::string Rijndael::Base::AlgorithmProvider() const
 	if (HasAESNI())
 		return "AESNI";
 #endif
-#if CRYPTOPP_SSE2_ASM_AVAILABLE && !defined(CRYPTOPP_DISABLE_RIJNDAEL_ASM)
+#if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE))
 	if (HasSSE2())
 		return "SSE2";
 #endif
@@ -530,18 +530,21 @@ void Rijndael::Base::UncheckedSetKey(const byte *userKey, unsigned int keyLen, c
 
 void Rijndael::Enc::ProcessAndXorBlock(const byte *inBlock, const byte *xorBlock, byte *outBlock) const
 {
-#if CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE) || CRYPTOPP_AESNI_AVAILABLE
-# if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE)) && !defined(CRYPTOPP_DISABLE_RIJNDAEL_ASM)
-	if (HasSSE2())
-# else
+#if CRYPTOPP_AESNI_AVAILABLE
 	if (HasAESNI())
-# endif
 	{
 		(void)Rijndael::Enc::AdvancedProcessBlocks(inBlock, xorBlock, outBlock, 16, 0);
 		return;
 	}
 #endif
 
+#if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE))
+	if (HasSSE2())	{
+		(void)Rijndael::Enc::AdvancedProcessBlocks(inBlock, xorBlock, outBlock, 16, 0);
+		return;
+	}
+#endif
+
 #if (CRYPTOPP_ARM_AES_AVAILABLE)
 	if (HasAES())
 	{
@@ -1255,7 +1258,7 @@ size_t Rijndael::Enc::AdvancedProcessBlocks(const byte *inBlocks, const byte *xo
 		return Rijndael_Enc_AdvancedProcessBlocks128_6x1_ALTIVEC(m_key, m_rounds, inBlocks, xorBlocks, outBlocks, length, flags);
 #endif
 
-#if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE)) && !defined(CRYPTOPP_DISABLE_RIJNDAEL_ASM)
+#if (CRYPTOPP_SSE2_ASM_AVAILABLE || defined(CRYPTOPP_X64_MASM_AVAILABLE))
 	if (HasSSE2())
 	{
 		if (length < BLOCKSIZE)
