diff --git a/src/cryptopp/CMakeLists.txt b/src/cryptopp/CMakeLists.txt
index 1dea197..f10fea5 100644
--- a/src/cryptopp/CMakeLists.txt
+++ b/src/cryptopp/CMakeLists.txt
@@ -310,15 +310,12 @@ endif()
 
 # ##############################################################################
 
-if(MSVC)
+if(WIN32)
   set(CRYPTOPP_MSVC_COMPILE_OPTIONS)
 
   # TODO: what about ICC and LLVM on Windows?
 
-  if(CMAKE_SYSTEM_VERSION MATCHES "10\\.0.*")
-    # https://docs.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt
-    list(APPEND CRYPTOPP_COMPILE_DEFINITIONS "_WIN32_WINNT=0x0A00")
-  endif()
+  list(APPEND CRYPTOPP_COMPILE_DEFINITIONS _WIN32_WINNT=0x0601 WINVER=0x0601 NTDDI_VERSION=0x06010000)
   # winapifamily.h is missing on AppVeyor machines
   include(CheckIncludeFileCXX)
   check_include_file_cxx("winapifamily.h" HAVE_WINAPIFAMILY_H)
