diff --git a/src/cryptopp/CMakeLists.txt b/src/cryptopp/CMakeLists.txt
index 1dea197..f10fea5 100644
--- a/src/cryptopp/CMakeLists.txt
+++ b/src/cryptopp/CMakeLists.txt
@@ -321,15 +321,12 @@ endif()
 
 # ##############################################################################
 
-if(MSVC)
+if(WIN32)
     set(CRYPTOPP_MSVC_COMPILE_OPTIONS)
 
     # TODO: what about ICC and LLVM on Windows?
 
-    if(CMAKE_SYSTEM_VERSION MATCHES "10\\.0.*")
-        # https://docs.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt
-        list(APPEND CRYPTOPP_COMPILE_DEFINITIONS "_WIN32_WINNT=0x0A00")
-    endif()
+    list(APPEND CRYPTOPP_COMPILE_DEFINITIONS _WIN32_WINNT=0x0601 WINVER=0x0601 NTDDI_VERSION=0x06010000)
     # winapifamily.h is missing on AppVeyor machines
     include(CheckIncludeFileCXX)
     check_include_file_cxx("winapifamily.h" HAVE_WINAPIFAMILY_H)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bb3bfbb..3309deb 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -247,26 +247,26 @@ else()
     endif()
 endif()
 
-if(Git_FOUND)
-    get_filename_component(
-        CRYPTOPP_CMAKE_PARENT_DIR
-        ${cryptopp-cmake_SOURCE_DIR}
-        DIRECTORY
-    )
-    execute_process(
-        COMMAND
-            ${CMAKE_COMMAND} -E env
-            GIT_CEILING_DIRECTORIES="${CRYPTOPP_CMAKE_PARENT_DIR}"
-            ${GIT_EXECUTABLE} rev-parse --abbrev-ref HEAD
-        WORKING_DIRECTORY ${cryptopp-cmake_SOURCE_DIR}
-        RESULT_VARIABLE cryptopp_CMAKE_GIT_RESULT
-        OUTPUT_VARIABLE cryptopp_GIT_BRANCH
-        ERROR_QUIET
-    )
-    if(cryptopp_CMAKE_GIT_RESULT EQUAL 0)
-        string(STRIP ${cryptopp_GIT_BRANCH} cryptopp_GIT_BRANCH)
-    endif()
-endif()
+#if(Git_FOUND)
+#    get_filename_component(
+#        CRYPTOPP_CMAKE_PARENT_DIR
+#        ${cryptopp-cmake_SOURCE_DIR}
+#        DIRECTORY
+#    )
+#    execute_process(
+#        COMMAND
+#            ${CMAKE_COMMAND} -E env
+#            GIT_CEILING_DIRECTORIES="${CRYPTOPP_CMAKE_PARENT_DIR}"
+#            ${GIT_EXECUTABLE} rev-parse --abbrev-ref HEAD
+#        WORKING_DIRECTORY ${cryptopp-cmake_SOURCE_DIR}
+#        RESULT_VARIABLE cryptopp_CMAKE_GIT_RESULT
+#        OUTPUT_VARIABLE cryptopp_GIT_BRANCH
+#        ERROR_QUIET
+#    )
+#    if(cryptopp_CMAKE_GIT_RESULT EQUAL 0)
+#        string(STRIP ${cryptopp_GIT_BRANCH} cryptopp_GIT_BRANCH)
+#    endif()
+#endif()
 if(NOT cryptopp_GIT_BRANCH)
     set(cryptopp_GIT_BRANCH "master")
 endif()
